config_defaults_inc.php

```diff
@@ -680,6 +680,82 @@
 $g_smtp_port = 25;
 
 /**
+ * Enable DomainKeys Identified Mail (DKIM) Signatures (rfc6376)
+ * To successfully sign mails you need to enable DKIM and provide at least:
+ * - DKIM domain
+ * - DKIM private key or key file path
+ * - DKIM selector
+ * - DKIM identity
+ * @see $g_email_dkim_domain
+ * @see $g_email_dkim_private_key_file_path
+ * @see $g_email_dkim_private_key_string
+ * @see $g_email_dkim_selector
+ * @see $g_email_dkim_identity
+ * @global integer $g_email_dkim_enable
+ */
+$g_email_dkim_enable = OFF;
+
+/**
+ * DomainKeys Identified Mail (DKIM) Signatures domain
+ * This is usually the same as the domain of your from email
+ * @see $g_from_email
+ * @see $g_email_dkim_enable
+ * @global string $g_email_dkim_domain
+ */
+$g_email_dkim_domain = 'example.com';
+
+/**
+ * DomainKeys Identified Mail (DKIM) Signatures private key path
+ * Path to the private key. If $g_email_dkim_private_key_string is specified
+ * this setting will not be used.
+ * @see $g_email_dkim_private_key_string
+ * @see $g_email_dkim_enable
+ * @global string $g_email_dkim_private_key_file_path
+ */
+$g_email_dkim_private_key_file_path = '';
+
+
+/**
+ * DomainKeys Identified Mail (DKIM) Signatures private key value
+ * This string should contain private key for signing. Leave empty
+ * string if you wish to load the key from the file defined with
+ * $g_email_dkim_private_key_file_path.
+ * @see $g_email_dkim_enable
+ * @see $g_email_dkim_private_key_file_path
+ * @global string $g_email_dkim_private_key_string
+ */
+$g_email_dkim_private_key_string = '';
+
+/**
+ * DomainKeys Identified Mail (DKIM) Signatures selector
+ * DNS selector for the signature (rfc6376)
+ * DNS TXT field should have for instance:
+ *   host mail.example._domainkey
+ *   value v=DKIM1; t=s; n=core; k=rsa; p=[public key]
+ * @see $g_email_dkim_enable
+ * @global string $g_email_dkim_selector
+ */
+$g_email_dkim_selector = 'mail.example';
+
+/**
+ * DomainKeys Identified Mail (DKIM) Signatures private key password
+ * Leave empty string if your private key does not have password
+ * @see $g_email_dkim_enable
+ * @global string $g_email_dkim_passphrase
+ */
+$g_email_dkim_passphrase = '';
+ 
+/**
+ * DomainKeys Identified Mail (DKIM) Signatures identity
+ * Identity you are signing the mails with (rfc6376)
+ * This is usually the same as your from email
+ * @see $g_from_email
+ * @see $g_email_dkim_enable
+ * @global string $g_email_dkim_identity
+ */
+$g_email_dkim_identity = 'noreply@example.com';
```

Comment:
Do we need this or should this `$g_from_email` be the one used to sign for the signing to match the from email address?

Reply:
If we need this as a separate config, then it should be added to `$g_public_config_names`.

Reply:
Yes, it is needed. Here is the spec, http://dkim.org/specs/rfc4871-dkimbase.html#rfc.section.1.1 . It clearly specifies that the signer is not required to be one of the headers of the mail. Signer, in general, can be other identity and then it is up to the receiving party to decide about the action. However, to void those kind of decisions it is good practice not to use a different one.

Reply:
How about we allow this to be empty and default to from_email or set its default value to `'%from_email%'`?

Reply:
Maybe, but to be honest like this the dkim domain, selector and identity are all working together to show the needed setup... if we would remove this value we would loose this connection.